src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Text;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+using Windows.Storage;
+using System.Collections.Generic;
+
+namespace Windows.Storage
+{
+	public partial class FileIO
+	{
+
+		private static async Task AppendWriteTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding, bool append)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			Encoding encodingForWriter = Encoding.UTF8;
+			switch (encoding)
+			{
+				case Streams.UnicodeEncoding.Utf8:
+					encodingForWriter = Encoding.UTF8;
+					break;
+				case Streams.UnicodeEncoding.Utf16LE:
+					encodingForWriter = Encoding.Unicode;
+					break;
+				case Streams.UnicodeEncoding.Utf16BE:
+					encodingForWriter = Encoding.BigEndianUnicode;
+					break;
+			}
+
+			using (Stream fileStream = await file.OpenStreamForWriteAsync())
+			{
+				if (append)
+				{
+					fileStream.Seek(0, SeekOrigin.End);
+				}
+				else
+				{
+					// reset file content, as UWP does
+					fileStream.SetLength(0);
+				}
+
+				using (StreamWriter streamWriter = new StreamWriter(fileStream, encodingForWriter))
+				{
+					await streamWriter.WriteAsync(contents);
+				}
+			}
+		}
+
+		public static IAsyncAction AppendTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding) => AppendWriteTextAsync(file, contents, encoding, true).AsAsyncAction();
+
+		public static IAsyncAction WriteTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding) => AppendWriteTextAsync(file, contents, encoding, false).AsAsyncAction();
+
+		private static string ConvertLinesToString(IEnumerable<string> lines)
+		{
+			string output = "";
+			foreach (string line in lines)
+			{
+				output = output + line + Environment.NewLine;
+			}
+
+			return output;
+		}
+
+		public static IAsyncAction WriteLinesAsync(IStorageFile file, IEnumerable<string> lines, Streams.UnicodeEncoding encoding)
+			=> WriteTextAsync(file, ConvertLinesToString(lines), encoding);
+
+		public static IAsyncAction AppendLinesAsync(IStorageFile file, IEnumerable<string> lines, Streams.UnicodeEncoding encoding)
+			=> AppendTextAsync(file, ConvertLinesToString(lines), encoding);
+
+
+		private static async Task AppendWriteTextAsync(IStorageFile file, string contents, bool append)
+		{
+			Streams.UnicodeEncoding unicodeEncoding = Streams.UnicodeEncoding.Utf8;
+
+			using (Stream fileReadStream = await file.OpenStreamForReadAsync())
+			{
+				using (BinaryReader binReader = new BinaryReader(fileReadStream))
+				{
+					try
+					{
+						byte firstByte = binReader.ReadByte();
+						switch (firstByte)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4827590)